### PR TITLE
[Postgres-Mssql-Oracle] Backport from (main->8.7)

### PR DESCRIPTION
**Pull request by jignesh-crest**
## Backport
Backports the GDC connector (Postgres, MsSQL, Oracle) from main to 8.7.

Note:
- Removed a test case from `test_generic_database.py` which was dependent on the latest version of the `AsyncIterator`. Updating this class will result in the failed test case of other tests from `test_byoc`.
- Added latest configuration changes from #618 and #621 [Please confirm if this change needs to be added in 8.7]

### CODE REVIEW

**Comment by moxarth-elastic** [comment]
buildkite test this

**Comment by moxarth-elastic** [comment]
buildkite test this

**Review by timgrein — COMMENTED** [review]
I don't think we need the changes for the configuration fields in 8.7. cc @navarone-feekery that's only planned for 8.8 or do we plan to backport rich-configurable fields?

**Comment by navarone-feekery** [comment]
@timgrein that's correct, there's no need to backport rich configurable fields. The previous structure should be fine for 8.7 (just `label`, `value`, and `type`).

**Comment by moxarth-elastic** [comment]
buildkite test this

**Comment by jignesh-crest** [comment]
> @jignesh-crest can you remove the changes from #618 and #621 as we won't have them in 8.7.

I've reverted changes in this commit [0bcf905](https://github.com/elastic/connectors-python/pull/630/commits/0bcf905658a5f9830f74765bfb5ddf84b7d3611f). Please let me know if any change is still pending to be reverted.

**Comment by jignesh-crest** [comment]
> > @jignesh-crest can you remove the changes from #618 and #621 as we won't have them in 8.7.
> 
> I've reverted the changes in this commit [0bcf905](https://github.com/elastic/connectors-python/pull/630/commits/0bcf905658a5f9830f74765bfb5ddf84b7d3611f). Please let me know if any change is still pending to be reverted.

@timgrein did you get a chance to review the changes?

File: docs/DEVELOPING.md [hunk 1]
```diff
@@ -612,6 +612,118 @@ $ make ftest NAME=oracle
...
+⚠️ _Currently, this connector is available in **beta** starting in 8.7_.
+Features in beta are subject to change and are not covered by the service level agreement (SLA) of features that have reached general availability (GA).
+
+Elastic versions 8.6.0+ are compatible with Elastic connector frameworks. Your deployment must include the Elasticsearch, Kibana, and Enterprise Search services.
+
+Microsoft SQL Server versions 2017, 2019, and Azure SQL, Amazon RDS for Microsoft SQL Server are compatible with Elastic connector frameworks.
+
+**Prerequisites**
+
+- The Microsoft SQL connector requires an ODBC driver to execute. Connector supports `Microsoft ODBC Driver 18`. To install ODBC driver please refer following [documentation](https://learn.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-driver-manager).
```
Comment (seanstory) [inline docs/DEVELOPING.md:630]: This is changing in https://github.com/elastic/connectors-python/pull/630. I think it's fine to temporarily backport this as is though, and backport the update from 630 when it's ready.

**Review by seanstory — APPROVED** [review]
LGTM.

**Pull request merged**